### DISCUSSION TITLE
使用mars3d鹰眼图失效的问题

### DISCUSSION THREAD

**Discussion by Eva-Skye** [zouyaoji/vue-cesium#254] [General]
![image](https://user-images.githubusercontent.com/45868259/184293762-6daae30a-bd0d-40dc-b582-04b02e7e0b6f.png)
如图所示，使用mars3d后，鹰眼图样式出现错误，不知是不是样式引用的问题。此外，方框内地图本应展示平面图，现在是一个球体。

**Comment by zouyaoji** [comment]
还没在第三方平台测个鹰眼组件，鹰眼组件实际上又起了个 vc-viewer，其他平台估计有bug，我周末测测修复一下。

**Comment by zouyaoji** [comment]
已在v3.0.17中修复。详见：https://zouyaoji.top/vue-cesium/#/zh-CN/component/platforms/vc-demo-mars3d
确实有bug，原因是 mars3d 初始化地球的方法与原生写法有较大差异，之前版本的写法 sceneMode 没传进去。
http://mars3d.cn/apidoc.html#Map，所以变成了三维球。
目前已经简单改成这种方式：
https://github.com/zouyaoji/vue-cesium/blob/dev/packages/components/viewer/src/useViewer.ts#L729

另外，其实并没有完全兼容 mars3d 的初始化地球的参数，如果想完全按照 mars3d 的 api  来初始化，vc-viewer 组件提供了 `viewerCreator`  属性，支持传入一个方法，由外部来创建 viewer 或者其他平台的实例，需要将创建结果的 viewer 对象返回。
https://github.com/zouyaoji/vue-cesium/blob/dev/packages/components/viewer/src/useViewer.ts#L726

**Reply by Eva-Skye** [reply]
惊叹于问题修复的速度，回复后三小时更新完，强

**Reply by zouyaoji** [reply]
正好周末~
